lib/index.js

```diff
@@ -45,13 +45,28 @@ Seal.prototype.move = function() {
   return this;
 }
 
+Seal.prototype.detectCollision = function(){
+  if (this.x > canvas.width){
+    this.speed = -1*this.speed;
+  } else if (this.x < 0){
+    this.speed = -1*this.speed;
+  }
+}
+
+
+
 Seal.prototype.draw = function() {
   context.fillRect(this.x, this.y, this.width, this.height)
 }
 
-Seal.prototype.turnAround = function(){
-  this.count --;
-
+Seal.prototype.turn = function(){
+  if (this.count > 0) {
+    this.count--;
+  }else if (this.count === 0){
+    //add new direction speed
+    this.count = (Math.floor(Math.random()*300)+100);
+  }
+  return this;
 }
```

Comment:
Added random turn counter that changes the direction of the seal based on a randomly generated counter